src/form-data-wrapper.js

```diff
@@ -121,11 +122,23 @@ class NativeFormDataWrapper {
   constructor(opt_form) {
     /** @private @const {!FormData} */
     this.formData_ = new FormData(opt_form);
+
+    if (opt_form) {
+      iterateCursor(opt_form.elements, input => {
+        if (input.type == 'file' && input.files.length == 0) {
+          this.formData_.delete(input.name);
```

Comment:
I'm worried this might be a breaking change if server expected `name` to be present for optional file input. 

Can we instead of removing it, make it whatever Chrome sends as the value of the name when dealing with empty files?

Reply:
I think that's a better call than what this PR currently does. I found someone else's workaround here and I think this PR should use the same strategy of a `File` with an empty `Blob` backing it.

Reply:
interesting solution. If issue is very specific to iOS 11 (fixed in 12?) let's also guard it to just iOS and 11 (or at least to just iOS)

Reply:
Sounds good, that should reduce risk